### PR TITLE
Fix daily GCS tests

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Fix daily GCS tests. Changes order in which monkeypatches are applied for test to mirror new format in GS object store

### CODE REVIEW

File: tests/utils/object_store/test_gs_object_store.py [hunk 1]
```diff
@@ -63,11 +63,13 @@ def test_gs_object_store_integration_json_auth(expected_use_gcs_sdk_val=True, cl
...
+def test_gs_object_store_integration_json_auth():
     with mock.patch.dict(os.environ):
-        if 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
-            del os.environ['GOOGLE_APPLICATION_CREDENTIALS']
-        test_gs_object_store_integration_json_auth(expected_use_gcs_sdk_val=False, client_should_be_none=True)
+        if 'GCS_KEY' in os.environ:
+            del os.environ['GCS_KEY']
+        if 'GCS_SECRET' in os.environ:
+            del os.environ['GCS_SECRET']
+        test_gs_object_store_integration_hmac_auth(expected_use_gcs_sdk_val=True, client_should_be_none=False)
```
Comment (snarayan21) [inline tests/utils/object_store/test_gs_object_store.py:72]: so iiuc there was some change in gcs sdk that changes the expected values here?

**Review by snarayan21 — APPROVED** [review]
seems calm, one question for personal clarity

**Comment by snarayan21** [comment]
@mvpatel2000 run dailies to check if the PR works?

**Comment by mvpatel2000** [comment]
@snarayan21 https://github.com/mosaicml/composer/actions/runs/9006991171

File: tests/utils/object_store/test_gs_object_store.py (hunk 1 above)
Comment (mvpatel2000) [reply tests/utils/object_store/test_gs_object_store.py:72]: We reordered which method is checked first

**Pull request merged**